tenzoMath/newRM.cpp

```diff
@@ -0,0 +1,62 @@
+#include "newRM.h"
+#define PI 3.14159265
+
+
+RoboModel::DhParameters::
+DhParameters(const double d, const double q, const double a, const double alpha)
+    : _dParam(d),
+      _qParam(q),
+      _aParam(a),
+      _alphaParam(alpha)
+{
+}
+
+
+RoboModel::RoboModel(std::vector<std::array<double, 4>> input)
+{
+    _kinematicChain.reserve(input.size());
+    for (int i = 0; i < input.size(); ++i)
+        _kinematicChain.push_back(DhParameters(input[i][0], input[i][1], input[i][2], input[i][3]));
+}
+
+RoboModel::~RoboModel()
```

Comment:
(**optional**): If you don't write your own constructor, you can use `~RoboModel() = default;` in class declaration.